frontend/data.ts

```diff
@@ -37,106 +40,25 @@ export async function createData(rep: Replicache) {
     return useSubscribe(rep, f, def);
   }
 
+  await rep.mutate.initClientState({
+    id: clientID,
+    defaultUserInfo,
+  });
+
   return {
     clientID,
 
     get rep(): Replicache {
       return rep;
     },
 
-    // mutators
-    createShape: rep.register(
-      "createShape",
-      async (tx: WriteTransaction, args: { id: string; shape: Shape }) => {
-        await putShape(writeStorage(tx), args);
-      }
-    ),
-
-    deleteShape: rep.register(
-      "deleteShape",
-      async (tx: WriteTransaction, id: string) => {
-        await deleteShape(writeStorage(tx), id);
-      }
-    ),
-
-    moveShape: rep.register(
-      "moveShape",
-      async (
-        tx: WriteTransaction,
-        args: { id: string; dx: number; dy: number }
-      ) => {
-        await moveShape(writeStorage(tx), args);
-      }
-    ),
-
-    resizeShape: rep.register(
-      "resizeShape",
-      async (tx: WriteTransaction, args: { id: string; ds: number }) => {
-        await resizeShape(writeStorage(tx), args);
-      }
-    ),
-
-    rotateShape: rep.register(
-      "rotateShape",
-      async (tx: WriteTransaction, args: { id: string; ddeg: number }) => {
-        await rotateShape(writeStorage(tx), args);
-      }
-    ),
-
-    initClientState: rep.register(
-      "initClientState",
-      async (
-        tx: WriteTransaction,
-        args: { id: string; defaultUserInfo: UserInfo }
-      ) => {
-        await initClientState(writeStorage(tx), args);
-      }
-    ),
-
-    setCursor: rep.register(
-      "setCursor",
-      async (
-        tx: WriteTransaction,
-        args: { id: string; x: number; y: number }
-      ) => {
-        await setCursor(writeStorage(tx), args);
-      }
-    ),
-
-    overShape: rep.register(
-      "overShape",
-      async (
-        tx: WriteTransaction,
-        args: { clientID: string; shapeID: string }
-      ) => {
-        await overShape(writeStorage(tx), args);
-      }
-    ),
-
-    selectShape: rep.register(
-      "selectShape",
-      async (
-        tx: WriteTransaction,
-        args: { clientID: string; shapeID: string }
-      ) => {
-        await selectShape(writeStorage(tx), args);
-      }
-    ),
-
-    deleteAllShapes: rep.register(
-      "deleteAllShapes",
-      async (tx: WriteTransaction) => {
-        await Promise.all(
-          (await tx.scanAll({ prefix: `shape-` })).map(([k]) => tx.del(k))
-        );
-      }
-    ),
+    ...rep.mutate,
```

Comment:
I kept this on data since that is what we pass around

Reply:
I think we can get rid of `Data` now though. The only reason for it was to implement this pattern that `Replicache` now does natively.